### ISSUE TITLE
:VimwikiFollowLink misbehaves with 'exclusive' selection option

### ISSUE THREAD

**Issue by louwers** [vimwiki/vimwiki#382]
To reproduce:

1. `set selection=exclusive`
2. insert `aaa bbb ccc ddd`, go to beginning of line
3. `v3e` (select three words)
4. `<CR>`

Expected behavior:

```
[[aaa bbb ccc]] ddd
```

Actual behavior:
```
[[aaa bbb ccc ]]ddd
```

**Comment by Nudin** [comment]
Dirty Hack to fix this:
https://github.com/vimwiki/vimwiki/commit/a2d64362bd0e5cb4d3b35508a7f625d4b12fbde3
Have to read into `selection` to find out proper way of handling this.

**Comment by louwers** [comment]
On StackOverflow this function is shared: https://stackoverflow.com/a/6271254/1929678

To be honest I think it should be part of Vim, I can ask the vim-dev mailing list.

**Comment by tinmarino** [comment]
Fixed : ac38bb2f09f

I created the function `vimwiki#u#get_selection(...)` to remove the hard to maintain hacks:
```
    " Save selected text to register "
    normal! gv""y
    call setreg('"', substitute(substitute(sub, '\n', '', ''), '\s', sc, 'g'), visualmode())
    normal! `>""pgvd
  finally
    call setreg('"', default_register_save, registertype_save)
    let &selection = sel_save
```

I used [this stackoverflow](https://stackoverflow.com/a/47051271/2544873) solution with the idea from the [first answer](https://stackoverflow.com/a/6271254/2544873)
```
" Get visual selection text content, optionaly replace its content
" :param: Text to replace selection
function! vimwiki#u#get_selection(...) abort

let lines[-1] = lines[-1][: column_end - (&selection == 'inclusive' ? 1 : 2)]
```
